src/VisualStudio/Core/Def/Implementation/FindReferences/StreamingFindUsagesPresenter.cs

```diff
@@ -275,5 +289,58 @@ protected static (Guid, string projectName, string? projectFlavor) GetGuidAndPro
 
             return (guid, projectName, projectFlavor);
         }
+
+        private void RemoveExistingInfoBar()
+        {
+            Contract.ThrowIfFalse(this.ThreadingContext.HasMainThread);
+
+            var infoBar = _infoBar;
+            _infoBar = null;
+            if (infoBar != null)
+            {
+                var infoBarHost = GetInfoBarHost();
+                infoBarHost?.RemoveInfoBar(infoBar);
+            }
+        }
+
+        private IVsInfoBarHost? GetInfoBarHost()
```

Comment:
would it be worth updating the existing info bar service to support infobars on various different windows as a parameter?

Reply:
Interesting idea. I can look into that!

Reply:
i looked into this and found practically nothing reusable here.